northstar/src/runtime/mod.rs

```diff
@@ -93,8 +86,8 @@ pub struct Runtime {
 }
 
 impl Runtime {
-    pub async fn start(config: Config) -> Result<Runtime, Error> {
-        let (stop_tx, stop_rx) = oneshot::channel();
+    pub async fn start(config: Config) -> Result<(Runtime, CancellationToken), Error> {
```

Comment:
One problem with `stop_wait` is that the runtime has to be moved to the tasks that handles the `SIGINT` and `SIGTERM`. In general, I think most of the time we just want to signal the shutdown without waiting for it. Separating the token from the runtime allows that. An other alternative that would be to send a `Request::Shutdown` instead of using the token.

Reply:
I think there's no need for a separate task at all as it's done in the current implementation. That a `Request::Shutdown` is sent internally is a impl detail and should not be visible on the pub API.

Reply:
Well separating these two things avoids to implement `Future` for `Runtime` and allows other type of "events" to trigger a shutdown if we so require in the future. I think this makes the handling of the runtime simpler and cleaner. Why is it better the way it was before?.